### PR TITLE
parser: limit error information size

### PR DESCRIPTION
**Pull request by tiancaiamao**
So the error message won't be too much, when the total sql is very long.

@shenli

### CODE REVIEW

File: parser/lexer.go [hunk 1]
```diff
@@ -80,6 +80,9 @@ func (s *Scanner) stmtText() string {
 func (s *Scanner) Errorf(format string, a ...interface{}) {
 	str := fmt.Sprintf(format, a...)
 	val := s.r.s[s.r.pos().Offset:]
+	if len(val) > 2048 {
+		val = val[:2048]
+	}
 	err := fmt.Errorf("line %d column %d near \"%s\"%s", s.r.p.Line, s.r.p.Col, val, str)
```
Comment (shenli) [inline parser/lexer.go]: Should we limit the length of str?

**Comment by coocood** [comment]
Print the length of the str if truncated.

**Comment by coocood** [comment]
LGTM

File: parser/lexer.go [hunk 2]
```diff
@@ -80,7 +80,10 @@ func (s *Scanner) stmtText() string {
 func (s *Scanner) Errorf(format string, a ...interface{}) {
 	str := fmt.Sprintf(format, a...)
 	val := s.r.s[s.r.pos().Offset:]
-	err := fmt.Errorf("line %d column %d near \"%s\"%s", s.r.p.Line, s.r.p.Col, val, str)
+	if len(val) > 2048 {
+		val = val[:2048]
+	}
+	err := fmt.Errorf("line %d column %d near \"%s\"%s (len %d)", s.r.p.Line, s.r.p.Col, val, str, len(s.r.s))
```
Comment (ngaut) [inline parser/lexer.go]: s/len/total length/

**Comment by tiancaiamao** [comment]
PTAL @ngaut @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**